### PR TITLE
feat(app): add cacheBuster and use URL internally

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Issue

Closes: AVO-166

## Description

This PR does two things, restructures the query functions to use URL objects internally and adds a cache busting function to the state and zone queries.

### Double check

- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: web/src/api/helpers.ts [hunk 1]
```diff
@@ -68,6 +68,18 @@ export function getBasePath() {
...
+  const current_date = new Date();
+  const date_to_hour = new Date(
+    current_date.getFullYear(),
+    current_date.getMonth(),
+    current_date.getDate(),
+    current_date.getHours(),
+    current_date.getMinutes()
+  );
+  return date_to_hour.toISOString();
+}
```
Comment (madsnedergaard) [inline web/src/api/helpers.ts:81]: Wouldn't it be simpler to do something like this instead?

```suggestion
export function cacheBuster(): string {
  const currentDate = new Date();
  currentDate.setSeconds(0);
  currentDate.setMilliseconds(0);
  
  return currentDate.toISOString();
}
```

**Review by madsnedergaard — COMMENTED** [review]
One comment, otherwise it looks good to me

File: web/src/api/helpers.ts (hunk 1 above)
Comment (VIKTORVAV99) [reply web/src/api/helpers.ts:81]: It would!

**Review by madsnedergaard — APPROVED** [review]
LGTM! This will not actually do anything until we switch the domain to `app-backend.electricitymaps.com` though, but might be worth doing that in a separate PR :)

**Pull request merged**